mod/peer/collections/policy/validator.go

```diff
@@ -7,24 +7,74 @@ SPDX-License-Identifier: Apache-2.0
 package policy
 
 import (
+	"fmt"
+
 	"github.com/hyperledger/fabric/protos/common"
+	"github.com/pkg/errors"
+	tdatapolicy "github.com/trustbloc/fabric-peer-ext/pkg/collections/transientdata/policy"
 )
 
-// Validator is a noop collection policy validator
+// Validator is a collection policy validator
 type Validator struct {
 }
 
-// NewValidator returns a noop collection policy validator
+// NewValidator returns a new collection policy validator
 func NewValidator() *Validator {
 	return &Validator{}
 }
 
 // Validate validates various collection config types
 func (v *Validator) Validate(collConfig *common.CollectionConfig) error {
-	return nil
+	config := collConfig.GetStaticCollectionConfig()
+	if config == nil {
+		return errors.New("unknown collection configuration type")
+	}
+
+	switch config.Type {
+	case common.CollectionType_COL_TRANSIENT:
+		return tdatapolicy.ValidateConfig(config)
+	default:
+		// Nothing to do
+		return nil
```

Comment:
It would be better if NewValidator didn’t return nil without a validity indicator.

Reply:
Not sure I know what you mean. Returning nil means that it is valid. This validator only handles extensions to the collection config. All of the basic checks have already been performed.

Reply:
Never mind :). All good.